### ISSUE TITLE
Add default_job_script to Adapter

### ISSUE THREAD

**Issue by ericfranz** [OSC/ood_job#5]
```ruby
attr_accessor :default_job_script
```

This would let me instantate an Adapter subclass such as:

```ruby
OodJob::Adapter::Torque.new(mycluster, default_job_script)
```

When submitting a job, if default_job_script is provided, this script would be merged in with the script that was passed in via `submit`.

**Comment by ericfranz** [comment]
The resource mgr's class name is `Torque`, so by convention we instantiate a new `Torque`. If we add optional "type" attribute to resource_mgr or server, this could be used as well to determine the adapter subclass to use by convention.

**Comment by nickjer** [comment]
I can do the same with (assuming `default_job_script` is a `Hash`)...

```ruby
# Using merge
script = OodJob::Script.new( default_job_script.merge(script.to_h) )

# Using double splat
script = OodJob::Script.new( **default_job_script, **script.to_h )
```

This I think has two benefits: 1. Developers can see explicitly what is happening to their script when they create their script. 2. We don't force all the adapters to handle the merging of the scripts in `#submit`.

There are other questions that arise from the initial issue posted: Is `default_job_script` an `OodJob::Script` object or a `Hash`? If it is an `OodJob::Script` object then it needs to be supplied `content` as that is a required attribute, so what should the `content` be?

Another option is the developer can create their own `ScriptDecorator` object:

```ruby
class ScriptDecorator < SimpleDelegator
  class << self
    attr_accessor :default_job_script
  end

  def initalize(opts)
    super OodJob::Script.new( self.class.default_job_script.merge(opts) )
  end
end
```

and use it as such...

```ruby
my_script = ScriptDecorator.new(content: "sleep 60", wall_time: 3600)
```

**Comment by ericfranz** [comment]
> 1. Developers can see explicitly what is happening to their script when they create their script
> 2. We don't force all the adapters to handle the merging of the scripts in #submit

I don't think either of these are relevant benefits. Currently we are hiding ood_job from developers in ood_job_rails so developers won't see it anyways. Similar to how you implemented `Array.wrap()`, there are simple ways to handle this so when implementing other adapters the impact is negligible.

However, the content attribute is a valid point.

**Comment by ericfranz** [comment]
This idea needs rethought. Closing.

**Comment by ericfranz** [comment]
Reopening this. This discussion is relevant and related to #6.

The issue that is OSC specific is setting the account string to the primary project of the user.  This is essentially an example of having some ruby code that is OSC specific that produces default values to be used in every job script that is submitted through an OOD app at OSC.

A related problem: we have talked about users being part of multiple projects, and being able to choose which project to submit jobs as. In this case, we probably need a different solution than setting a value globally; unless we have an OOD-wide user config that the user can modify.

Other attributes that would be nice to set defaults for: number of nodes and ppn

**Comment by nickjer** [comment]
I am fine with a default job script that a sys admin can administer, but the code that injects it into a job submission may be better placed in `ood_job_rails` rather than `ood_job`. As the use-case being that it will be supplied somehow globally by a sys admin, and `ood_appkit` will probably be responsible with reading/parsing this information that `ood_job_rails` depends on.

Another solution which I feel may be better option is piggybacking off of what system's already provides the users with the qsub submit_filters. So the global config of the yaml could take the form of:

```yaml
#/etc/ood/config/clusters.d/ruby.yml
...
        resource_mgr:
          type: "OodCluster::Servers::Torque"
          data:
            host: "ruby-batch.osc.edu"
            lib: "/opt/torque/lib64"
            bin: "/opt/torque/bin/ruby" # <-- note, this is cluster specific
            version: "6.0.1"
...
```

And there will exist a submit_filter (or wrapper) called `/opt/torque/bin/ruby/qsub` that Puppet will be responsible for updating that sets default values submit attributes that are not already defined for a job being submitted to the Ruby cluster.

Systems already uses these for setting default memory, file system requirements, ... that are not specified in the batch job. So why not ask them to tack on a few more cluster specific options like default account name.

Pros:
  - gets around us having to add ruby code for determining the account name (which can have strange mappings for certain sys admin users) in all apps
  - gets around having to determine a yaml format for all the job submission attributes (e.g., ppn)
  - systems is responsible for maintaining it

Cons:
  - not sure how difficult or easy it would be for systems to implement

**Comment by nickjer** [comment]
More can be read here:

http://docs.adaptivecomputing.com/torque/5-1-3/help.htm#topics/torque/12-appendices/jobSubmissionFilter.htm%3FTocPath%3DAppendices%7C_____10

**Comment by nickjer** [comment]
Maybe we just request a custom webnode SUBMITFILTER script that handles what we want as defaults?

**Comment by ericfranz** [comment]
Yes. Though an ood-wide user config that is user-modifiable and can change environment variables or provide values available to each of the apps is something that would be great to have. Think of the "Settings" app in iOS, that have a centralized location of settings for each app.
